sklearn/tests/test_calibration.py

```diff
@@ -400,20 +400,14 @@ def test_calibration_curve():
     y_true = np.array([0, 0, 0, 1, 1, 1])
     y_pred = np.array([0.0, 0.1, 0.2, 0.8, 0.9, 1.0])
     prob_true, prob_pred = calibration_curve(y_true, y_pred, n_bins=2)
-    prob_true_unnormalized, prob_pred_unnormalized = calibration_curve(
```

Comment:
Although `normalize` is deprecated, we are still supporting it until 1.3. This means that tests for `normalize` should remain intact. For the warnings, we can decorate the test function with a `filterwarning`:

```python
# TODO(1.3): Remove filterwarning and tests for normalize
@pytest.mark.filterwarnings("ignore::FutureWarning:sklearn")
def test_calibration_curve():
    """Check calibration_curve function"""
    y_true = np.array([0, 0, 0, 1, 1, 1])
    y_pred = np.array([0.0, 0.1, 0.2, 0.8, 0.9, 1.0])
    prob_true, prob_pred = calibration_curve(y_true, y_pred, n_bins=2)
    prob_true_unnormalized, prob_pred_unnormalized = calibration_curve(
        y_true, y_pred * 2, n_bins=2, normalize=True
    )
    ...
```

Note that the body of the function stays the same as `main`.

Reply:
This is a good point, and was how I had originally structured the PR. Based on a discussion with @glemaitre, we established that it might be more favorable to adjust the test such that all coverage is retained for the `normalize` parameter under its own unit test.

In this way, removing the test for `normalize` in 1.3 is a bit more straightforward because one can simply remove `test_calibration_curve_with_unnormalized_proba`

Reply:
Ah I see. Thank you for the explanation. I am happy with this change.